### ISSUE TITLE
Missing NDK support for static runtimes/no runtime

### ISSUE THREAD

**Issue by mtak-** [facebook/buck#741]
There's no way to specify static or shared runtime (always shared) in the `[ndk]` section of `.buckconfig`.

We currently rely on `libc++`; however, there's a [bug](http://developer.android.com/ndk/guides/cpp-support.html#rc) in `c++_shared` where exceptions do not always work on arm devices. The solution is to use `c++_static`, but this is not possible using `cxx_library` rules as they are now. A fix for this would be much appreciated.

EDIT: A more detailed explanation of the exceptions issue is [here](http://gosuwachu.io/2016/04/09/android-ndk-bug/)

**Comment by Coneko** [comment]
cc @andrewjcg

**Comment by mtak-** [comment]
If anyone else needs to use a static lib, here's a workaround (YMMV)
`.buckconfig`:
```
... other settings ...

[ndk]
    ... other ndk settings ...
    cxx_runtime = system

... more config settings ...
```

`BUCK` file:
```python
cxx_library(
    ### other settings

    compiler_flags = [
        ### other compiler flags
        '-I**PATH/TO/NDK**/sources/cxx-stl/llvm-libc++/libcxx/include',
        '-I**PATH/TO/NDK**/sources/cxx-stl/llvm-libc++abi/libcxxabi/include',
        '-I**PATH/TO/NDK**/sources/android/support/include',
    ],
    linker_flags = [
        ### other linker flags
        '-Wl,--whole-archive',
        '-l**PATH/TO/NDK**/sources/cxx-stl/llvm-libc++/libs/**ARCH**/libc++_static.a',
        '-Wl,--no-whole-archive',
    ]
)

**Comment by hey99xx** [comment]
Is this the official solution to the problem or is there any cleaner way? 

We should have double amount of options in https://buckbuild.com/concept/buckconfig.html#ndk.cxx_runtime since all runtimes come with shared and static variations.

**Comment by styurin** [comment]
This was implemented in 3dddf893259dae0e4a492d29f4c5ecb575d96810
